README.md

```diff
@@ -161,14 +170,17 @@ See [C++ Programmer's Toolbox](#c-programmers-toolbox) for details.
 
 1.  **Parameters**
     *   Ordering: Input and then Output.
-    *   All **references** must be **const**, and this is the recommended
-        way to pass parameters.
-    *   For output parameters, pass pointers.
+    *   Either use const references or pointer for non-fundamental types. The stl types are not fundamental types.
+    *   Use **const reference** for read only parameter. Use **pointer** for writable parameter.
+    
 2.  **Default Argument**
-    *   Not recommended for readability issue. **Do not use** unless
+    *   Not recommended for readability/ABI issue. **Do not use** unless
         you have to.
+
 3.  **Trailing Return Type Syntax**
     *   When in lambda. Period.
+    *   When using template, and the output type should be deducted by input types.
```

Comment:
s/by/from/

I think this is a good place to have an example :) Our readers may not be equally experienced in C++, and an examples may speak to them better ~